network/peer/tls_config_test.go

```diff
@@ -0,0 +1,82 @@
+// Copyright (C) 2019-2024, Ava Labs, Inc. All rights reserved.
+// See the file LICENSE for licensing terms.
+
+package peer_test
+
+import (
+	"crypto/ecdsa"
+	"crypto/elliptic"
+	"crypto/rand"
+	"crypto/rsa"
+	"crypto/tls"
+	"crypto/x509"
+	"errors"
+	"testing"
+
+	"github.com/stretchr/testify/require"
+
+	"github.com/ava-labs/avalanchego/network/peer"
+)
+
+func TestValidateRSACertificate(t *testing.T) {
+	for _, testCase := range []struct {
+		description string
+		input       func() tls.ConnectionState
+		expectedErr error
+	}{
+		{
+			description: "Valid TLS cert",
+			input: func() tls.ConnectionState {
+				key, err := rsa.GenerateKey(rand.Reader, 2048)
+				require.NoError(t, err)
+				x509Cert := makeRSACertAndKey(t, key)
+				return tls.ConnectionState{PeerCertificates: []*x509.Certificate{&x509Cert.cert}}
+			},
+		},
+		{
+			description: "No TLS certs given",
+			input: func() tls.ConnectionState {
+				return tls.ConnectionState{}
+			},
+			expectedErr: errors.New("no certificates sent by peer"),
```

Comment:
any particular reason you don't want to use the error object you've already been allocating ? ( ErrNoCertsSent )